### ISSUE TITLE
include bug

### ISSUE THREAD

**Issue by KonoromiHimaries** [gorhill/uBlock#3685]
if i use `!#include /headers/adblock_headery.txt` in

```
https://github.com/KonoromiHimaries/polish-ads-filter/blob/master/internal-filters/adblock_wewnetrzne.txt
```

then i have double slash
![bez tytulu](https://user-images.githubusercontent.com/5884000/38450545-073832c2-3a20-11e8-9244-1d50e1186663.png)

ubo 1.15.24
chrome 65.0.3325.181

**Comment by krystian3w** [comment]
Maybe no use `/`

**Comment by WowDude** [comment]
I assume that he already tried before creating the issue.

**Comment by gwarser** [comment]
Relative path `/headers/adblock_headery.txt` inside `https://github.com/KonoromiHimaries/polish-ads-filter/blob/master/internal-filters/adblock_wewnetrzne.txt` should give `https://github.com/headers/adblock_headery.txt` UBO is not supporting this https://github.com/gorhill/uBlock/issues/3331#issuecomment-354554069

**Comment by krystian3w** [comment]
When delete first `/` link should ok and uBlock load lists.

**Comment by WowDude** [comment]
1. I've just checked and both `//` and` /` works - the sublist is loaded and works, so it's just a typo/cosmetic bug, not a functional bug, which of course doesn't change the fact that it should be repaired.
2. The orginal issue that was reproted by @KonoromiHimaries was just as the above (1) - one unnecessary backslash `/`
3. Now @gwarser is reporting another - different issue, and as far as I see, the comment he linked to says:

> from with the same directory

So relative path 
`/headers/adblock_headery.txt` inside 
`https://github.com/KonoromiHimaries/polish-ads-filter/blob/master/internal-filters/adblock_wewnetrzne.txt` should give not 
`https://github.com/headers/adblock_headery.txt` like @gwarser said but 
`https://github.com/KonoromiHimaries/polish-ads-filter/blob/master/internal-filters/headers/adblock_headery.txt` like @KonoromiHimaries said because this is:
**`https://github.com/KonoromiHimaries/polish-ads-filter/blob/master/internal-filters/`** the same directory, and this: `https://github.com/` is just the same domain and not the directory.

@gwarser perhaps you messed up directory with domain? As per https://github.com/gorhill/uBlock/issues/3331#issuecomment-355750880 gorhill answered about directory https://github.com/gorhill/uBlock/issues/3331#issuecomment-355751233 and not about domain.

**Comment by gorhill** [comment]
> it's just a typo/cosmetic bug

I don't consider this a bug.

**Comment by BilBg** [comment]
> Suggestion 1:
> #3331 (comment)
> 
> !#include ../path2/includedfile.txt

That "relative path" usage was suggested in a draft by AdguardTeam (to go to parent directory)
https://github.com/AdguardTeam/AdguardBrowserExtension/issues/917

My post with `Test_include_relative_path.txt` was to show that this can be used to `!#include` non-own file.

> If a sublist contains the substring .., it will be rejected.

By "it" I think @gorhill means: **the entire `!#include` line will be rejected** (and not only the `..`)

> Suggestion 2:
> #3331 (comment):
> 
> !#include /easylist/easylist/master/easylist/easylist_whitelist.txt

That was not a "Suggestion" but a question - to ensure that the new code dealing with `..` can't be fooled to start from root by using `/` as first symbol.

Note:
Sometimes there is no problem with using many `///`(the same page is opened)
https://github.com/gorhill///uBlock/
https://github.com/gorhill////uBlock/
https://github.com/gorhill///////uBlock/

https://www.youtube.com////////////////watch?v=O47KDrMBtSk&t=3
https://www.youtube.com/watch?v=O47KDrMBtSk&t=3

... but not always:
https://www.wikiwand.com/en/Krisia_Todorova
https://www.wikiwand.com/en//Krisia_Todorova

https://pastebin.com/raw/dHidQupm
https://pastebin.com/raw//dHidQupm

**Comment by WowDude** [comment]
> That "relative path" usage was suggested in a draft by AdguardTeam (to go to parent directory)
> AdguardTeam/AdguardBrowserExtension#917
> 
> My post with Test_include_relative_path.txt was to show that this can be used to !#include non-own file.

I know that it wasn't suggested by you, as it is quoted...and I didn't want to suggest that it was your suggestion, I just failed to find where the quoted part does come from, if I could find it, I would quote the original source (comment), probably I was busy and just missed it, so I just linked to your comment, but I meant what inside the quoted part was a suggestion, and not you/your comment just because it contains the quote.

> 
> >  If a sublist contains the substring `..`, it will be rejected.
> 
> > By "it" I think @gorhill means: the entire !#include line will be rejected (and not only the ..)

See..."I think that it refers to a", "he thinks that it refers to b", "she thinks that it refers to c"..., everybody thinks different, it should be clear what "it" refers to, too much guesswork what "it" refers to, that's why it leads to misinterpretations, but most probably you're right anyway, and btw. does the comma mean that `rejected` was reffering to `sublist` and not to `..`> ? 😛

> That was not a "Suggestion" but a question

Let it stay a suggestive question... 😛

**Comment by BilBg** [comment]
> If a sublist contains the substring `..`, it will be rejected.

Maybe should be:
If a sublist **filename/path** contains the substring `..`,  the  `!#include` will be rejected.

@gorhill - Is there a way for the author of the list to know what is rejected/ignored?
Or generally if `!#include` failed, e.g. if sublist do not exist (typo in filename/path)?

**Comment by krystian3w** [comment]
> Is there a way for the author of the list to know what is rejected/ignored? 

Show in uBlock (counter / source list).

> ```
> ! >>>>>>>> full url
> !-----------------------Your filters -----------------------!
> ! . . .
> ! <<<<<<<< full url
> ```

No see this elements when sublist is rejected/ignored.

If user report ads to your project you may please see list or ask if he sees an advertisement on any site (sublist blocking it already).

---

> Or generally if !#include failed, e.g. if sublist do not exist (typo in filename/path)?

Yes, list no load and no block elements/actions/communication/files.

**Comment by krystian3w** [comment]
@gorhill  multiple use `#include` is acceptable / correct?

When I try use 3 `#include`, last `#include` duplicate sublist on main list and show counter with new not used filters.
Duplicate last list have when update other list or disable other - most often.

**Comment by gorhill** [comment]
> multiple use #include is acceptable / correct?

Yes. If you see an issue, please provide steps to reproduce. Also, mind that #3624 was fixed in 1.15.20.
